src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs

```diff
@@ -77,7 +77,12 @@ private void NoteDeclaredPatternVariables(BoundPattern pattern)
             if (IsInside && pattern.Kind == BoundKind.DeclarationPattern)
             {
                 var decl = (BoundDeclarationPattern)pattern;
-                _variablesDeclared.Add(decl.LocalSymbol);
+                if (decl.Variable.Kind == SymbolKind.Local)
```

Comment:
It depends how this walker handles Deconstruction Declaration. It shouldn't try to add fields into _variablesDeclared list. You should definitely consider adding tests for region analysis for Deconstruction Declaration.

---
In reply to: [80753230](https://github.com/dotnet/roslyn/pull/13999#discussion_r80753230) [](ancestors = 80753230)